### PR TITLE
Rewrite test cases with pytest

### PR DESCRIPTION
**Pull request by ender503**
Since the test cases are written in shell script and Python,
it's hard to maintain them.

To eliminate this problem, rewrite the test cases in pytest and
name the test case file with prefix "test_".

Some of API does not response JSON data correctly, so fix them.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ $ make
 
 (Optional) run test suite:
 ```shell
-$ make check
+$ pytest
```
Comment (jserv) [inline README.md]: Can you invoke `pytest` via `make check` as well?

File: tests/test_core.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import os
```
Comment (jserv) [inline tests/test_core.py]: The files in directory `tests` don't have to follow the prefix "test_" in their name.

Comment (jserv) [reply tests/test_core.py]: Instead, something like `tests/core.py` and `ext-tangleid.py` makes sense.

Comment (ender503) [reply tests/test_core.py]: This looks better 👍

File: README.md (hunk 1 above)
Comment (ender503) [reply README.md]: I change back to `make check`. 👌

**Comment by ender503** [comment]
@jserv, ping? 🤔

File: Makefile [hunk 3]
```diff
@@ -12,21 +12,8 @@ $(DCURL_LIB): $(DCURL_DIR)
...
-	echo "Running test suite..." ; \
-	( python $^ & echo $$! > $${TMP_PID} ); \
-	sleep 3 ; \
-	for i in $(TESTS); do \
-	    ( echo "\n\n==[ $$i ]==\n"; $$i || kill -9 `cat $${TMP_PID}` ) \
-	done ; \
-	kill -9 `cat $${TMP_PID}`
-	@$(RM) $${TMP_PID}
-
+check:
```
Comment (jserv) [inline Makefile]: Ensure the dependency for build procedure.

File: tests/core.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        "type": TIPS_TYPE_TWO_NULL_IOTA_TNX,
+    }
+    rv = client.post('/', data=json.dumps(post_data))
+    response = json.loads(rv.data)
+
+    assert len(response['trunkTransaction']) is 81
+    assert len(response['branchTransaction']) is 81
+
+def test_send_transfer(client):
+    tag = "PYTHONTEST"
```
Comment (jserv) [inline tests/core.py]: It should be specific to this package such as "SWARMTEST".

Comment (ender503) [reply tests/core.py]: Replace it with "SWARMTEST". 👍

File: Makefile (hunk 3 above)
Comment (ender503) [reply Makefile]: Replace it with "check: server.py $(DCURL_LIB)". 👍

**Pull request merged**